sbe-tool/src/test/resources/field-order-check-schema.xml

```diff
@@ -0,0 +1,281 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<sbe:messageSchema xmlns:sbe="http://fixprotocol.io/2016/sbe"
+                   package="order.check"
+                   id="1"
+                   description="Unit Test"
+                   byteOrder="littleEndian"
+                   version="1">
+    <types>
+        <composite name="messageHeader" description="Message identifiers and length of message root">
+            <type name="blockLength" primitiveType="uint16"/>
+            <type name="templateId" primitiveType="uint16"/>
+            <type name="schemaId" primitiveType="uint16"/>
+            <type name="version" primitiveType="uint16"/>
+            <type name="numGroups" primitiveType="uint16"/>
+        </composite>
+        <composite name="groupSizeEncoding" description="Repeating group dimensions">
+            <type name="blockLength" primitiveType="uint8"/>
+            <type name="numInGroup" primitiveType="uint8"/>
+        </composite>
+        <composite name="varDataEncoding" semanticType="Length">
+            <type name="length" primitiveType="uint8" semanticType="Length"/>
+            <type name="varData" primitiveType="char" semanticType="data"/>
+        </composite>
+        <composite name="point" semanticType="Point">
+            <type name="x" primitiveType="int32"/>
+            <type name="y" primitiveType="int32"/>
+        </composite>
+        <enum name="direction" encodingType="int8">
+            <validValue name="BUY">1</validValue>
+            <validValue name="SELL">2</validValue>
+        </enum>
+        <type name="ipV4Address" primitiveType="uint8" length="4"/>
+        <set name="flags" encodingType="uint8">
+            <choice name="guacamole">1</choice>
+            <choice name="cheese">2</choice>
+            <choice name="sourCream">4</choice>
+        </set>
+        <type name="ccyPair" primitiveType="char" characterEncoding="US-ASCII" length="6"/>
+    </types>
+
```

Comment:
I would like to see one additional case of a data field within a nested group, i.e:
```
    <sbe:message name="GroupAndVarLength" id="2">
        <field name="a" id="1" type="int32"/>
        <group name="b" id="2">
            <field name="c" id="3" type="int32"/>
            <group name="b" id="2">
                <field name="c" id="3" type="int32"/>
                <data name="d" ...
```

Reply:
:+1: I'll add some tests for that.

Reply:
Added tests for this case in c0b8cdb.